From ab64cd261a81a7e472476cdf133b8d1a176815f4 Mon Sep 17 00:00:00 2001
From: Zheng Yang <zhengyang@rock-chips.com>
Date: Fri, 22 May 2015 18:34:17 +0800
Subject: [PATCH] HDMI: register display type according lcdc property.

	If property of lcdc connected by HDMI is EXTERN,
	we register HDMI as a aux display device.

Signed-off-by: Zheng Yang <zhengyang@rock-chips.com>
---
 drivers/video/rockchip/hdmi/rockchip-hdmi-core.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/video/rockchip/hdmi/rockchip-hdmi-core.c b/drivers/video/rockchip/hdmi/rockchip-hdmi-core.c
index 4c65868f8397..4b6cef9480bb 100644
--- a/drivers/video/rockchip/hdmi/rockchip-hdmi-core.c
+++ b/drivers/video/rockchip/hdmi/rockchip-hdmi-core.c
@@ -470,12 +470,15 @@ struct hdmi *rockchip_hdmi_register(struct hdmi_property *property,
 	hdmi->yscale = 100;
 	hdmi_init_modelist(hdmi);
 
-#ifndef CONFIG_ARCH_RK29
 	if (hdmi->property->videosrc == DISPLAY_SOURCE_LCDC0)
 		hdmi->lcdc = rk_get_lcdc_drv("lcdc0");
 	else
 		hdmi->lcdc = rk_get_lcdc_drv("lcdc1");
-#endif
+
+	if (hdmi->lcdc->prop == EXTEND)
+		hdmi->property->display = DISPLAY_AUX;
+	else
+		hdmi->property->display = DISPLAY_MAIN;
 	memset(name, 0, 32);
 	sprintf(name, "hdmi-%s", hdmi->property->name);
 	hdmi->workqueue = create_singlethread_workqueue(name);
-- 
2.35.3

